conan/tools/google/bazeldeps.py

```diff
@@ -229,19 +235,90 @@ def load_conan_dependencies():
             )
         {% endfor %}
         """)
+    module_template = textwrap.dedent("""\
+        # This module provides a repo for each requires-dependency in your conanfile.
+        # It's generated by the BazelDeps, and should be used in your Module.bazel file.
+        load(":conan_deps_repo_rules.bzl", "conan_dependency_repo")
+
+        def _load_dependenies_impl(mctx):
+        {% for repository_name, pkg_folder, pkg_build_file_path in dependencies %}
+            conan_dependency_repo(
+                name = "{{repository_name}}",
+                package_path = "{{pkg_folder}}",
+                build_file_path = "{{pkg_build_file_path}}",
+            )
+        {% endfor %}
+
+            return mctx.extension_metadata(
+                root_module_direct_deps = None,
```

Comment:
If you set this to `"all"`, Bazel will show a warning if the user doesn't `use_repo` all repos defined by this extension and `bazel mod tidy` updates `use_repo` automatically. If I understood the logic correctly, that's exactly what you want here.

Reply:
@fmeum thanks for the explanation! Then, it makes total sense 👍

Reply:
Not sure, if it really makes sense. Some of the dependencies listed in the conanfile, you might not want to use in the bazel-build, or?

Reply:
If there is a way to detect which direct dependencies should be used, those could also be returned as a list here. Could also be a follow-up though.

Reply:
For me, the suggestion of "all" would be a great usability improvement. I am sure when developers add something to conanfile they will forget to add it to the MODULE.bazel. The warning and possible automation would help a lot.

For us, we will only have things in conanfile that we want in the build; I'm not sure why you would list something there and then not use it?

Reply:
Yeah, agree.
I was just thinking about generator packages or compiler-packages, for me it is still not really clear how this will work.

Reply:
Yeah, I got what you meant @Neeeflix Let me check it locally and see how it behaves 😁

Reply:
I was running some tests, and it's indeed useful to see something like this:

```bash
WARNING: /Users/franchuti/develop/conan/bazel/test_package/MODULE.bazel:1:40: The module extension conan_extension defined in @//conan:conan_deps_module_extension.bzl reported incorrect imports of repositories via use_repo():

Not imported, but reported as direct dependencies by the extension (may cause the build to fail):
    build-cmake, zlib

Fix the use_repo calls by running 'bazel mod tidy'.
```
I did not find any side effects. IMO, let's give it a try. Thanks to everyone 😁